tikv/pd
# api: use easyjson and context in regions interface

### What problem does this PR solve?


Issue Number: Close https://github.com/tikv/pd/issues/6835

### What is changed and how does it work?



```commit-message
```

### Check List



Tests



- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

1. bench
` go test -benchmem -run=^$ -bench ^BenchmarkGetRegions$ github.com/tikv/pd/server/api -benchtime=30s`

| | times |  ns/op | B/op | allocs/op |
|:-:|:-:|:-:|:-:|:-:|
| default | 12    |    2964076204|      1153885878 | 4253342 |
| default |  12   |       2985435560 |         1271459127  | 4261952  | 
| default |12      |   2994243983  |        1241259906  |4255758 |
| jsoniter| 15  |  2492527400 |   2384667309 | 4044502| 
| jsoniter| 18    |     2428125878  |        2545873184  | 4018365 |
| jsoniter|18   |      2207472486  |      2214775903  | 3925291 |
| easyjson| 26   |    1381140866 |       803330238 |    3582216 | 
| easyjson|28   |   1339109264  |         786423035  |    3564425  | 
| easyjson|26     |     1370970701  |      801957518  |   3575059  | 
|this pr(c501fd8)| 24     |   1386651999  |         789385029  |   3577663  | 
|this pr(c501fd8)| 24     |    1512373970 |   814244526  |  3632067  |
|this pr(c501fd8)|  26   |      1455024327 |     794162807 |   3603735 | 
|this pr(2b1ab88)| 31   |     1343236075|        558050128 | 3568320 |
|this pr(2b1ab88)| 30        |    1249315290|   551981506|  3534328|
|this pr(2b1ab88)|  31        |      1316071881|     544873733|   3557758 | 

  | ns/op | ns diff | B/op | Bytes diff | allocs/op | alloc diff
-- | -- | -- | -- | -- | -- | --
default | 2981251916 |   | 1222201637 |   | 4257017 |  
jsoniter | 2376041921 | -20% | 2381772132 | 95% | 3996053 | -6%
easyjson | 1363740277 | -54% | 797236930 | -35% | 3573900 | -16%
this pr(c501fd8) | 1451350099 | -51% | 799264121 | -35% | 3604488 | -15%
this pr(2b1ab88) | 1302874415| -56% | 551635122| -55% | 3553468| -17%



2. kill and ensure the goroutine exit
client: `curl -s -o /dev/null http://192.168.8.88:2379/pd/api/v1/regions ` and ctrl-c
server: goroutine exit

### Release note



```release-note
None.
```
